warp10/src/main/java/io/warp10/script/functions/TOENCODER.java

```diff
@@ -50,6 +57,40 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
     GTSEncoder encoder = new GTSEncoder(0L);
     
     for (Object element: elements) {
+      // If we encounter a GTS, a wrap or raw wrap, add all of it to the encoder
+
+      // If wrap, convert to raw wrap
+      if (element instanceof String) {
+        element = OrderPreservingBase64.decode(element.toString().getBytes(Charsets.US_ASCII));
+      }
+
+      // If raw wrap, convert to GTS
+      if (element instanceof byte[]) {
+        TDeserializer deser = new TDeserializer(new TCompactProtocol.Factory());
+
+        try {
+          GTSWrapper wrapper = new GTSWrapper();
+
+          deser.deserialize(wrapper, (byte[]) element);
+
+          element = GTSWrapperHelper.fromGTSWrapperToGTS(wrapper);
```

Comment:
It would be better to get a GTSDecoder rather than a GeoTimeSeries, this would prove faster and less hungry in memory